runtime/metricsview/executor.go

```diff
@@ -127,10 +127,40 @@ func (e *Executor) ValidateQuery(qry *Query) error {
 	panic("not implemented")
 }
 
-// Watermark returns the current watermark of the metrics view.
-// If the watermark resolves to null, it defaults to the current time.
-func (e *Executor) Watermark(ctx context.Context) (time.Time, error) {
-	return e.loadWatermark(ctx, nil)
+// Timestamps queries min, max and watermark for the metrics view
+func (e *Executor) Timestamps(ctx context.Context, executionTime *time.Time) (time.Time, time.Time, time.Time, error) {
```

Comment:
How about creating a result type here? Like:
```
type TimestampsResult struct {
  Min time.Time
  Max time.Time
  Watermark time.Time
}
```

Then it can also be used instead of the `min, max, watermark time.Time` in `type Executor`